### ISSUE TITLE
Test release 2.45

### ISSUE THREAD

**Issue by cesine** [OpenSourceFieldlinguistics/FieldDB#1931]
@farahabbasi and @hisakonog  what are your schedules like in the coming week, are you available for another round or two of testing, any time between April 1-8?

Background of this release:
Some of the field methods group were complaining that the spreadsheet app couldn't handle a lot of data. We were hoping to avoid porting the spreadsheet to the commonjs, but since we dont want to put teams on dative yet, and even if we did between now and before the end of the semester we wouldnt have time to test  to feel like it was ready. As a result, last week we finished porting the functionality of the entire backbone library to the commonjs library. 

The result is we have another major release coming up for commonjs and angular core components  which is now powering both the spreadsheet (hisako) and the psycholing dashboard (farah).The Commonjs itself has good testing coverage, nearly 3000 assertions now, but there are many things in the user interface that aren't testable in node js, or we haven't written tests for yet (audio recording, file upload, changing focus and auto-glossing etc) so we really need humans to test, and probably in a few iterations. 

* Spreadsheet
In order to paginate the data loading, we refactored the original code so it is using the commonjs  and angular core to load, save and edit models, resulting in pagination to make the app faster. This impacts every aspect of the app, so every aspect must be tested. We also upgraded to Twitter bootstrap 3 so the buttons now have a brighter look which means the Sikuli tests might need new screenshots.

├──Spreadsheet angular app uses:
└──FieldDB angular components which uses:
    └── FieldDB common JS core

* Psycholing dashboard
The most important thing was that we removed the HTML5 routing to use the normal # routing since the app was rather broken when running on the server under Apache. We also updated to v2.49 of the commonjs with automatic authentication resume if the user's couch token is still live. Import is also now using objects per cell rather than just a string, permitting a more complex and non-lossy import. Numbers and dates are auto-detected and displayed with date picker number input box etc. This means  the import features, and the login and routing functionality would be the focus of the testing.

├──Pyshcholing dashboard app uses:
└──FieldDB angular components which uses:
    └── FieldDB common JS core

├──SAILS MontageJS app uses:
└──OPrime MontageJS components which uses:
    └── FieldDB common JS core

├──TCPP MontageJS app uses:
└──OPrime MontageJS components which uses:
    └── FieldDB common JS core

├──TDFP MontageJS app uses:
└──OPrime MontageJS components which uses:
    └── FieldDB common JS core

├──TDFM MontageJS app uses:
└──OPrime MontageJS components which uses:
    └── FieldDB common JS core

**Comment by farahabbasi** [comment]
HI Gina,

I will be free and available by april 4. Will that be ok? This week at work
is slightly demanding with end of the month deadlines. Are you back?

**Comment by hisakonog** [comment]
yes I am able to do the testing april 1-9.

**Comment by cesine** [comment]
@hisakonog  and @farahabbasi in our meeting on March 31st we cut the phophlo branch of the project. i've only been able to work on it about 6 days in the past 3 months and i don't think i will have time in the future to support it's users once it goes into production. dr rvachew found some funding to work on it so that it can get done faster.

this means we don't need @farahabbasi  to test version 2.49 in phophlo, instead we can take our time with it and release it for the learn-x users when its ready. i wasn't able to completely refactor the spreadsheet in under 2 weeks, so I think we should focus efforts on dative instead so that the spreadsheet can be replaced rather than updated to handle pagination.

I deployed v2.45.3 to appdev, and its ready for @hisakonog 's testing, main differences to test:
* remove roles from users 

But you should test everything else too.

**Comment by farahabbasi** [comment]
Can you tell me what I need to test now? I'm not sure I am working as per
requirement...

**Comment by hisakonog** [comment]
- [ ] learn x can download sample data
 - [x] load the app (http://appdev.lingsync.org)
 - [x] login
 - [ ] if browser (firefox) remembers password user can just click login
 - [ ] register
 - [ ] create a corpus
 - [ ] rename a corpus
 - [ ] edit a corpus description and view the changes in hovering over the corpus in the corpora list
 - [x] select corpus
 - [ ] select lingllama's corpus
 - [ ] change to another corpus
 - [ ] it should show the corpus title and current session on the black nav bar when editing data
 - [ ] enter 1 datum
 - [ ] delete 1 datum
 - [ ] keyboard focus always goes first field in edit template (when you use teh arrows to go between datum)
 - [ ] up and down arrows goes between datum and between pages once you are in a datum
 - [ ] enter in a new datum brings you to the next datum, in a normal datum it does nothing
 - [ ] new datum are showing on the page you are on, until you use a pagination arrow,then they fall into the chronological order they were entered
 - [ ] tab takes you between fields in the data
 - [ ] hover over a field tells you the teams help conventions of that field
 - [ ] optional grammatical judgement shows up in front of utterance
 - [ ] ~~refresh page and see if fields for McGill template show up (Utterance, Morphemes, Gloss, Translation, IPA, Notes)~~
 - [ ] use full template
 - [ ] ~~load the app uses the fulltemplate regardless of the local storage preferences~~
 - [ ] use full template with user specified fields
 - [ ] ~~use mcgill template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] ~~use yale template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] ~~use compact template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] use pagination arrows
 - [ ] save 1 or many datum
 - [ ] edit 1 or many datum in the same version
 - [ ] edit 1 or many datum from a previous version
 - [ ] editing a datum should produce only one activity and only one modified by users field
 - [ ] create a new session
 - [ ] app selects most recent session by default
 - [ ] edit a session
 - [ ] delete a session
 - [ ] record audio 
 - [ ] ~~record video~~ 
 - [ ] upload an mp3 
 - [ ] upload a wav
 - [ ] upload a mp4 (or any movie)
 - [ ] add a comment
 - [ ] search for something
 - [ ] search for t 
 - [ ] search for z 
 - [ ] click on export
 - [ ] the new fields (ipa and notes) are exported correctly
 - [ ] export a lot of data
 - [ ] enter wrong password
 - [ ] enter wrong username
 - [ ] change password
 - [ ] reset password (incorrect passwords 5 times)
 - [ ] use the forgot password button (and follow its instructions)
 - [ ] try connecting to server when user is offline
 - [ ] try connecting to server when server is offline
 - [ ] test glosser
 - [ ] others...

**Comment by hisakonog** [comment]
Login & choose corpus OK, chose a session but unable to see the data: the screen remained the same with the session dropdown. Clearing the cache solved the problem.

**Comment by hisakonog** [comment]
Saving seems to take longer than before (> 30 sec). Not sure if it is due to the actual network connection or to something else (I'm currently on the school network). Error message shows up even though I seem to have a good enough connection... 
![screen shot 2015-04-13 at 16 00 53](https://cloud.githubusercontent.com/assets/760694/7124516/ae56c0d4-e1f7-11e4-81b8-0bd696b78bba.png)

**Comment by hisakonog** [comment]
- [x] change to another corpus 
When I left one corpus (say corpus A) open and unattended for more than 0.5h then switched to another corpus (corpus B), the corpus name showed Corpus B but the session dropdown showed sessions of the corpus A. This happened twice. The first time there was a modal appeared but I couldn't see what the message was (I was switching btwn windows and probably pressed return or something). Second time I refreshed the page and the app logged me out.

**Comment by hisakonog** [comment]
Search mode: 

There is "select all" button but we don't have "clear selected" button. Do we want it? 
There is "clear search result" button but when I clicked that to clear the selection, I got timed out. 
![screen shot 2015-04-14 at 16 52 48](https://cloud.githubusercontent.com/assets/760694/7148200/d3d03c8c-e2ce-11e4-87bb-f9474ce4ca83.png)
![screen shot 2015-04-14 at 16 52 30](https://cloud.githubusercontent.com/assets/760694/7148208/e7e55266-e2ce-11e4-99cc-ffea6823569d.png)

**Comment by hisakonog** [comment]
April 15 14:33  Login as nemo33 OK, but fetching some corpora seems to have failed (it says internal server error 500). 

![screen shot 2015-04-15 at 14 33 26](https://cloud.githubusercontent.com/assets/760694/7166695/6bea41d0-e37d-11e4-8343-0973b72d0054.png) 
![screen shot 2015-04-15 at 14 34 52](https://cloud.githubusercontent.com/assets/760694/7166712/8069c270-e37d-11e4-88dd-48aa1a4512bd.png)

Logout button didn't log me out. 

![screen shot 2015-04-15 at 14 35 34](https://cloud.githubusercontent.com/assets/760694/7166802/27962a70-e37e-11e4-8923-fddc4749197d.png)
![screen shot 2015-04-15 at 14 36 09](https://cloud.githubusercontent.com/assets/760694/7166804/2d108266-e37e-11e4-8ec6-6ec596cd30fb.png)

Closed (and left the page) and reopened the app, logged in as nemo33, then all the corpora were fetched properly.

**Comment by hisakonog** [comment]
- [ ] remove someone from corpus

I REMOVED nemo77 (used to be readonly) from malayalam corpus, then got this message... 

![screen shot 2015-04-15 at 16 38 46](https://cloud.githubusercontent.com/assets/760694/7168813/37ef09d6-e38e-11e4-92fb-53ee40f56f85.png)

**Comment by hisakonog** [comment]
- [x] add someone to a corpus 

As Joel pointed out before the Berkeley workshop, the spinning ball spins for quite a long time (roughly a minute or more) before the notification modal appears. The newly added user receives an email notification immediately after the admin clicks "Add" button, though. So I guess it is the  GUI which responds slow somehow. 

I think the second clause of the notification message should be removed. 
"User X now has (type of access permission) access to (corpus name) ~~, the user was already a member of this corpus team~~."

**Comment by hisakonog** [comment]
- [ ] editing a datum should produce only one activity and only one modified by users field 

Expanding a datum without any edition/modification still records activity. It seems that expanding a datum while the "Save" button is RED produces this behaviour.

**Comment by cesine** [comment]
Hi @hisakonog

I went through all the issues you found, and found the source for all except https://github.com/OpenSourceFieldlinguistics/FieldDB/issues/1931#issuecomment-93077948 where the server is timing out. I tried to improve this a bit by redcuing the number of map reduces in the corpora which are re-indexed when the spreadsheet loads, but this probably will have very little effect and we are stuck with this problem of things timing out until we switch dative to be more efficient too, and switch users to dative. 

Can you re-test again? 
In thies relese (spreadhseet v2.45.04ss) 

Basically, these are the sections which changed but this kind of means that everything in the checkboxes needs to be rechecked.
* changes to datum saving
* changes to activity saving
* changes to session loading
* changes to new session
* changes to session saving
* changes to session editing
* changes to corpus saving
* changes to corpus loading
* changes to corpora list loading
* changes to the glosser loading

**Comment by cesine** [comment]
hi @hisakonog i saw you logging in on the auth serer. its deployed by the way (around 6am your time) so we need to test this version and release it/fix it as soon as we can. the new auth service is mostly backward compatable (lizzie is also working right now, and i also tested using the old apps) but it woudl be bet to get everything released together. i'm falling asleep. can you man the support emails tonight if any one files a bug report?

**Comment by hisakonog** [comment]
v2.45.04.ss

 - [ ] learn x can download sample data
 - [x] load the app (http://appdev.lingsync.org)
 - [x] login
 - [ ] if browser (firefox) remembers password user can just click login
 - [ ] register
 - [ ] create a corpus
 - [x] rename a corpus
 - [x] edit a corpus description and view the changes in hovering over the corpus in the corpora list
 - [x] select corpus
 - [x] select lingllama's corpus
 - [x] change to another corpus
 - [x] it should show the corpus title and current session on the black nav bar when editing data
 - [x] enter 1 datum
 - [x] delete 1 datum
 - [x] keyboard focus always goes first field in edit template (when you use the arrows to go between datum)
 - [x] up and down arrows goes between datum and between pages once you are in a datum
 - [x] enter in a new datum brings you to the next datum, in a normal datum it does nothing
 - [x] new datum are showing on the page you are on, until you use a pagination arrow,then they fall into the chronological order they were entered
 - [x] tab takes you between fields in the data
 - [x] hover over a field tells you the teams help conventions of that field
 - [x] optional grammatical judgement shows up in front of utterance
 - [ ] ~~refresh page and see if fields for McGill template show up (Utterance, Morphemes, Gloss, Translation, IPA, Notes)~~
 - [ ] ~~use full template~~
 - [ ] ~~load the app uses the fulltemplate regardless of the local storage preferences~~
 - [ ] use full template with user specified fields
 - [ ] ~~use mcgill template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] ~~use yale template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] ~~use compact template (no longer needed, corpus admins can customize the interface themselves)~~
 - [x] use pagination arrows
 - [x] save 1 or many datum
 - [x] edit 1 or many datum in the same version
 - [x] edit 1 or many datum from a previous version
 - [ ] editing a datum should produce only one activity and only one modified by users field
 - [x] create a new session 
 - [x] change sessions 
 - [ ] app selects most recent session by default
 - [x] edit a session
 - [x] delete a session
 - [ ] record audio 
 - [ ] ~~record video~~ 
 - [ ] upload an mp3 
 - [ ] upload a wav
 - [ ] upload a mp4 (or any movie)
 - [x] add a comment
 - [ ] search for something
 - [x] search for t 
 - [x] search for z 
 - [x] click on export
 - [ ] the new fields (ipa and notes) are exported correctly
 - [x] export a lot of data
 - [x] enter wrong password
 - [x] enter wrong username
 - [x] change password
 - [x] reset password (incorrect passwords 5 times)
 - [x] use the forgot password button (and follow its instructions)
 - [ ] try connecting to server when user is offline
 - [ ] try connecting to server when server is offline
 - [ ] test glosser
 - [ ] remove someone from a corpus 
 - [x] add someone to a corpus 
 - [ ] others...

**Comment by hisakonog** [comment]
- [x] rename a corpus
 - [x] edit a corpus description and view the changes in hovering over the corpus in the corpora list

To see the edit of corpus name/description in the corpora list you need to either logout then re-login or refresh page.  But refreshing the corpora list page shows the data entry page of the most recent corpus/session.

**Comment by hisakonog** [comment]
- [ ] add a new field 
 - [x] use full template with user specified fields

Clicking "Add a new field" button on the corpus settings page does nothing. 
Changed the title of an existing field (not the ones predefined in the app, but the one I had added in the previous version), which did not work earlier (sometime between 11:00-12:00, 20/04/2015, I got the error saving corpus details, see below) but is working now as of 14:50, 20/04/2015.  

![changefieldlabel](https://cloud.githubusercontent.com/assets/760694/7238039/c3111206-e76d-11e4-97b8-ebcc6b167bc0.png)

**Comment by hisakonog** [comment]
- [x] create a new session  

11:47 20/04/2015  I was unable to create a new session in the Session mode ("Create session" button did nothing, see below), but was able to do it in the Corpora list page. 

![createnewsession](https://cloud.githubusercontent.com/assets/760694/7238186/8f110a50-e76e-11e4-96db-4ad3324b82fb.png)

Later, I tried to create a session in the Session mode again, and this time it worked. It doesn't seem to be the new version per se, but maybe something to do with the connection to the server.

**Comment by hisakonog** [comment]
- [ ] Search for something 

Strange thing: Searched for a string "56" and the search hit datums that do not contain "56" 

![search56-1](https://cloud.githubusercontent.com/assets/760694/7238900/d4f890a6-e773-11e4-9e3f-10542cfe9d84.png)
![search56-2](https://cloud.githubusercontent.com/assets/760694/7238909/eb33cc14-e773-11e4-97c4-0e61eae96f55.png)

Searching for "5" hits all the 15 records in the session, searching for "6" hits 13 (tztztz and tt tt tt t are not hit by search 6)

**Comment by hisakonog** [comment]
- [x] delete a session 

I thought "Delete Session" button only appears when a session is empty? 

![deletesession](https://cloud.githubusercontent.com/assets/760694/7239371/6bb26e06-e777-11e4-866d-a2e947668208.png)

**Comment by hisakonog** [comment]
@cesine  I got debugger "Error loading data" many times when testing. It was either vendor.js 105156 or 24130 .

**Comment by cesine** [comment]
@hisakonog 
new version is v2.45.05ss

* triing a new thing for server timeouts
* found and fixed adding new field to corpus
* found and fixed the corpus details not refreshing right away
* delete session can be done any time, no need to empty it. if you delete it, it will not show in your list but the data is not actually affected in anyway. you can find the data if you search. i don tkno wwhy you could only delete empty sesison before anyway, if we dont provide a way to batch delete or batch move this doesnt really make sense. 
* searching for 5 will get alld atum in made in 2015. i think this is okay...

**Comment by hisakonog** [comment]
LingSync v2.45.05ss 

 - [ ] learn x can download sample data
 - [x] load the app (http://appdev.lingsync.org)
 - [x] login
 - [ ] if browser (firefox) remembers password user can just click login
 - [ ] register
 - [ ] create a corpus
 - [] rename a corpus
 - [] edit a corpus description and view the changes in hovering over the corpus in the corpora list
 - [x] select corpus
 - [] select lingllama's corpus
 - [x] change to another corpus
 - [x] it should show the corpus title and current session on the black nav bar when editing data
 - [x] enter 1 datum
 - [x] delete 1 datum
 - [] keyboard focus always goes first field in edit template (when you use the arrows to go between datum)
 - [x] up and down arrows goes between datum and between pages once you are in a datum
 - [x] enter in a new datum brings you to the next datum, in a normal datum it does nothing
 - [x] new datum are showing on the page you are on, until you use a pagination arrow,then they fall into the chronological order they were entered
 - [x] tab takes you between fields in the data
 - [x] hover over a field tells you the teams help conventions of that field
 - [x] optional grammatical judgement shows up in front of utterance
 - [ ] ~~refresh page and see if fields for McGill template show up (Utterance, Morphemes, Gloss, Translation, IPA, Notes)~~
 - [ ] ~~use full template~~
 - [ ] ~~load the app uses the fulltemplate regardless of the local storage preferences~~
 - [x] add a new field 
 - [x] use full template with user specified fields
 - [ ] ~~use mcgill template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] ~~use yale template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] ~~use compact template (no longer needed, corpus admins can customize the interface themselves)~~
 - [x] use pagination arrows
 - [x] save 1 or many datum
 - [x] edit 1 or many datum in the same version
 - [x] edit 1 or many datum from a previous version
 - [ ] editing a datum should produce only one activity and only one modified by users field
 - [] create a new session 
 - [] change sessions 
 - [] app selects most recent session by default
 - [] edit a session
 - [] delete a session
 - [] record audio 
 - [ ] ~~record video~~ 
 - [] upload an mp3 
 - [] upload a wav
 - [ ] upload a mp4 (or any movie)
 - [] add a comment
 - [ ] search for something
 - [] search for t 
 - [] search for z 
 - [] click on export
 - [] the new (user defined) fields ~~(ipa and notes)~~ are exported correctly
 - [] export a lot of data
 - [] enter wrong password
 - [] enter wrong username
 - [] change password
 - [] reset password (incorrect passwords 5 times)
 - [] use the forgot password button (and follow its instructions)
 - [ ] try connecting to server when user is offline
 - [ ] try connecting to server when server is offline
 - [ ] test glosser
 - [ ] remove someone from a corpus 
 - [] add someone to a corpus 
 - [ ] others...

**Comment by hisakonog** [comment]
Strange thing happened. I choose a "to be deleted" session and it showed the dummy datums (from previous entry) that are in the tbd session, so me without thinking start entering more dummy data for test. Then trying to change sessions, it turned out that the current session I thought I was in was in fact session for negation, not session tbd. I realised that even though the screen was showing datums from tbd, the session name on the nav bar was negation. So I went to session tbd, there was no datums I just entered. I went to neg session and there they are my dummy datums. It happened between 16:08 - 16:16 04/052015. I keep the dummies just in case @cesine wants to take a look. 
I even changed corpus and UI settings during these 8 mins. Oh, then I refreshed the page to have the new UI in effect. Could that be a reason?

**Comment by hisakonog** [comment]
Created a new user on appdev. His "default" fields are like this: 

![screen shot 2015-05-05 at 7 01 04 am](https://cloud.githubusercontent.com/assets/760694/7471429/d9248a18-f2f4-11e4-8070-a945cab98b32.png)
The same "default" template as Jessica had. Not sure where it is from, maybe not just due to the incompatibility of auth and v2.45.02.01.33ss.  

(Sidenote: also had problem logging in the new user 6:47am, 6:55am, error msg: unable to contact production server.  Restarted chrome and opened appdev, the user was already logged in.)

**Comment by hisakonog** [comment]
- [ ] edit a corpus description and view the changes in hovering over the corpus in the corpora list

Corpus description edit does not immediately show in the hover-over in the corpora list, in order to see the changes you need to re-login.

**Comment by hisakonog** [comment]
testing  v2.45.05ss as real TEST user (so I don't get timeouts, hopefully...) 
Registration OK. 
Using practice corpus, entered some data. Changed Corpus and UI settings. 
- Save button in corpus setting responds slow, you need to click twice with some interval (10sec < ?) between clicks in order to save the change 
- for UI setting change to take effect need to refresh page. then I got "error finding corpus details in nemotest1-firstcorpus" 
![after page refresh](https://cloud.githubusercontent.com/assets/760694/7479326/f18b3f9a-f32e-11e4-9b6b-eb6534d3f2bc.png)
 
got logged out and re-logged in, practice corpus is invisible... 
![screen shot 2015-05-05 at 14 26 47](https://cloud.githubusercontent.com/assets/760694/7479940/01c6513e-f333-11e4-8c09-1a84254cf376.png)


logged out again and re-login then things seems fine... Practice corpus is visible and i have data that had been entered.  maybe I hit the server hiccup time?

**Comment by hisakonog** [comment]
@cesine the checked items in the above (go 5 comments up) are tested and seem to be working fine. 
It seems that the "Save" button on the corpus field settings responds rather slow and I get "Range error: Maximum call stuck size exceeded". It usually works when I click Save the second time, after some interval. See 2 comments up.

**Comment by cesine** [comment]
@hisakonog its deployed as v2.45.06ss on appdev for your testing. all of the issues in this milestone should be ready except the help texts

**Comment by hisakonog** [comment]
LingSync v2.45.06ss 

 - [ ] learn x can download sample data
 - [ ] load the app (http://appdev.lingsync.org)
 - [ ] login
 - [ ] if browser (firefox) remembers password user can just click login
 - [ ] register
 - [ ] create a corpus
 - [ ] rename a corpus
 - [ ] edit a corpus description and view the changes in hovering over the corpus in the corpora list
 - [ ] select corpus
 - [ ] ~~select lingllama's corpus~~
 - [ ] change to another corpus
 - [ ] it should show the corpus title and current session on the black nav bar when editing data
 - [ ] enter 1 datum
 - [ ] delete 1 datum
 - [ ] keyboard focus always goes first field in edit template (when you use the arrows to go between datum)
 - [ ] up and down arrows goes between datum and between pages once you are in a datum
 - [ ] enter in a new datum brings you to the next datum, in a normal datum it does nothing
 - [ ] new datum are showing on the page you are on, until you use a pagination arrow,then they fall into the chronological order they were entered
 - [ ] tab takes you between fields in the data
 - [ ] hover over a field tells you the teams help conventions of that field
 - [ ] optional grammatical judgement shows up in front of utterance
 - [ ] ~~refresh page and see if fields for McGill template show up (Utterance, Morphemes, Gloss, Translation, IPA, Notes)~~
 - [ ] ~~use full template~~
 - [ ] ~~load the app uses the fulltemplate regardless of the local storage preferences~~
 - [ ] add a new field 
 - [ ] use full template with user specified fields
 - [ ] ~~use mcgill template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] ~~use yale template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] ~~use compact template (no longer needed, corpus admins can customize the interface themselves)~~
 - [ ] use pagination arrows
 - [ ] save 1 or many datum
 - [ ] edit 1 or many datum in the same version
 - [ ] edit 1 or many datum from a previous version
 - [ ] editing a datum should produce only one activity and only one modified by users field
 - [ ] create a new session 
 - [ ] change sessions 
 - [ ] app selects most recent session by default
 - [ ] edit a session
 - [ ] delete a session
 - [ ] record audio 
 - [ ] ~~record video~~ 
 - [ ] upload an mp3 
 - [ ] upload a wav
 - [ ] upload a mp4 (or any movie)
 - [ ] add a comment
 - [ ] search for something
 - [ ] search for t 
 - [ ] search for z 
 - [ ] click on export
 - [ ] the new (user defined) fields ~~(ipa and notes)~~ are exported correctly
 - [ ] export a lot of data
 - [ ] enter wrong password
 - [ ] enter wrong username
 - [ ] change password
 - [ ] reset password (incorrect passwords 5 times)
 - [ ] use the forgot password button (and follow its instructions)
 - [ ] try connecting to server when user is offline
 - [ ] try connecting to server when server is offline
 - [ ] test glosser
 - [ ] remove someone from a corpus 
 - [ ] add someone to a corpus 
 - [ ] others...

**Comment by hisakonog** [comment]
- [ ] load the app (http://appdev.lingsync.org)  

Error message when loading the app. But after clicking ok, it seems to work normally. 

![screen shot 2015-05-19 at 11 30 08](https://cloud.githubusercontent.com/assets/760694/7706810/8d25105c-fe1a-11e4-9f88-5205e3b6d155.png) 

zooming in the error at localhost:6984
![screen shot 2015-05-19 at 11 32 52](https://cloud.githubusercontent.com/assets/760694/7706859/d9409f88-fe1a-11e4-98db-d800011cf717.png)

**Comment by hisakonog** [comment]
- [ ] register 

Registering seemed to have  gone smoothly, but after logging in as the newly created user, I can't access the first corpus. "Saving" button keeps blinking. 

![screen shot 2015-05-19 at 11 41 21](https://cloud.githubusercontent.com/assets/760694/7707080/19c116b8-fe1c-11e4-816f-ce9db6a76ebf.png)

![screen shot 2015-05-19 at 11 47 24](https://cloud.githubusercontent.com/assets/760694/7707239/e59aa812-fe1c-11e4-856c-28c6bec45530.png)

![screen shot 2015-05-19 at 11 48 10](https://cloud.githubusercontent.com/assets/760694/7707262/f5c2bd56-fe1c-11e4-98ba-319bb3fb9195.png)

**Comment by hisakonog** [comment]
another screenshot for select corpus problem

![screen shot 2015-05-20 at 9 32 45 pm](https://cloud.githubusercontent.com/assets/760694/7739976/2cc5c408-ff38-11e4-9aaf-adeb082845c7.png)

**Comment by hisakonog** [comment]
- [x] edit a corpus description and view the changes in hovering over the corpus in the corpora list

Still need to refresh page to see the edit of the corpus name and description in the list

**Comment by hisakonog** [comment]
- [ ] use pagination arrows

Pagination arrows don't show up in the app.

**Comment by hisakonog** [comment]
- [ ] user-defined field names are showing in the Template preview in the corpus setting 

IN the template preview the help text updates when you edit it, but the field name doesn't.

**Comment by hisakonog** [comment]
Now the version is 3.4. Moving to #1987 and closing this.
